tools/c7n_left/c7n_left/providers/terraform/graph.py

```diff
@@ -76,10 +80,11 @@ def __init__(self):
 
     @staticmethod
     def is_id_ref(v):
-        if len(v) != 36:
-            return False
-        if v.count("-") != 4:
+        try:
+            uuid.UUID(hex=v)
```

Comment:
I wonder if this is going to be significantly slower, due to the exception catch, as we have to sniff all of the string values.

Reply:
So, I did a quick `timeit` test comparing the original version, this one and a third one using a regexp. It does seem that the UUID version is quite slow, but it's actually the `UUID()` creation that takes the most time, not the exception handling.

These are the times on my machine:

check_orig: 0.06528643200484414
check_uuid: 0.6836905783372155
check_re: 0.16567468399686427

Maybe the regexp case (`r"[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$"`) is a reasonable compromise?